enhancements/microshift/rhoai-model-serving-on-microshift.md

```diff
@@ -0,0 +1,339 @@
+- Install kserve
+- Create an InferenceService using some basic model that can be easily handled by CPU
+  - Model shall be delivered using ModelCar approach (i.e. model inside an OCI container)
+- Make a HTTP call to the InferenceService to verify it is operational
+- Make a HTTP call to the InferenceService's `/metrics` endpoint
+  (possibly before and after an inference request)
+
+This simple scenario will assert basic integration of the features:
+that everything deploys and starts correctly - sort of quick and easy sanity test.
+See [upstream kserve's "First InferenceService" guide](https://kserve.github.io/website/master/get_started/first_isvc/)
+which represents similar verification.
+
+Another test that should be implemented is using a hardware with an NVIDIA GPU,
+for example AWS EC2 instance. Goal is to assert that all the elements in the
+stack will work together on MicroShift. The test itself should not be much
+different from the sanity test (setup everything, make a call to the
+InferenceServing), but it should leverage a serving runtime and a model that
+require a GPU.
+Implementation of this test will reveal any additional dependencies such as
+device plugin, drivers, etc.
+This kind of test can run periodically - once or twice a week initially,
+with frequency adjusted later if needed.
+
+EC2 instance type candidates (ordered - chepeast first in us-west-2):
+- g4dn.xlarge (4 vCores 2nd gen Intel Xeon, 16 GiB, NVIDIA T4 GPU 16GiB)
+- g6.xlarge (4 vCores 3rd gen AMD EPYC, 16 GiB, NVIDIA L4 GPU 24GiB)
+- g5.xlarge (4 vCores 2nd gen AMD EPYC, 16 GiB, NVIDIA A10G GPU 24GiB)
+- g6e.xlarge (4 vCores 3rd gen AMD EPYC, 16 GiB, NVIDIA L40S GPU 48GiB)
+
+## Graduation Criteria
+
+### Dev Preview -> Tech Preview
+
+RHOAI's kserve on MicroShift will begin Tech Preview.
+
+### Tech Preview -> GA
+
+Advancement to GA depends on RHOAI's support for Raw Deployments.
+
+### Removing a deprecated feature
+
+N/A
+
+## Upgrade / Downgrade Strategy
+
+By having RHOAI's kserve in MicroShift spec they will share version, so it's
+expected that in case of MicroShift upgrade, the kserve also will be upgraded.
+
+MicroShift team might need to monitor (or simply test upgrades) version changes
+of the kserve CRDs, so there's always an upgrade path.
+
+MicroShift team will need to monitor (or automate) RHOAI versions to make sure
+each MicroShift release uses most recent stable RHOAI version (relates to the
+MicroShift rebase procedure - RHOAI not being part of OCP payload is sort of
+side loaded).
+
+## Version Skew Strategy
+
+MicroShift and kserve RPMs built from the same .spec file should not introduce
+a version skew.
+
+## Operational Aspects of API Extensions
+
+N/A
+
+## Support Procedures
+
+For the most part, RHOAI's and/or kserve support procedures are to be followed.
+
+Although there might some cases where debugging MicroShift might be required.
+One example is Ingress and Routes, as this is the element that kserve will
```

Comment:
What about endpoint security?
In RHOAI we use ServiceMesh + Authorino in the Serverless configuration while for RawDeployment we expect to use oauth-proxy (it will be replaced by something else that has not been defined yet)

Reply:
Good point! I think that needs to be configurable by the user. And I can imagine serveral ways.

It could be that the client is just another pod in another namespace, running "locall". That would either require
1. no auth at all
2. based on a service account 

But it could also be "external" clients, and  for those
3.  then anything OAuth based (e.g. oauth-proxy) totally makes sense to me. Just be aware that MicroShift itself does not have a built in OAuth provider (like OpenShift has). So one would need to point the proxy to an external auth provider (e.g. keycloak, MicroSoft).

Does that make sense?

Reply:
Yes I think it makes totally sense and overall I don't know if this is a common pattern for MicroShift components or not so if there is a common pattern to adopt I would probably follow that (i.e. external provider or whatever)